src/Core/Composer/ModulesConfigurationProcessor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Composer;
+
+use Composer\IO\IOInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+final class ModulesConfigurationProcessor
+{
+    /**
+     * @var IOInterface the CLI IO interface
+     */
+    private $io;
+
+    public function __construct(IOInterface $io)
+    {
+        $this->io = $io;
+    }
+
+    /**
+     * Foreach module, will install the module dependencies
+     * into /modules/{module}/vendor folder.
+     *
+     * @param array $configuration the Composer configuration
+     * @param string $rootPath the modules directory location
+     */
+    public function processInstallation(array $configuration, $rootPath)
+    {
+        $composerExecutable = $rootPath.'/vendor/composer/composer/bin/composer';
```

Comment:
It's weird that you use `/vendor` concatenated to `$rootPath` when you initially create `$rootPath` from `$rootPath = realpath($composer->getConfig()->get('vendor-dir'). '/..');`
I don't think you actually need `$rootPath`, you'd better use 'vendor-dir' from the config when you need to access a vendor such as composer, and inject `modules-dir` rather than include it in the config (is their a purpose in allowing to modify it in the config, since the core expect modules to be in `modules` anyway)